src/TestSuiteResult.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Humbug;
+
+use Humbug\Utility\CoverageData;
+use Symfony\Component\Process\PhpProcess;
+
+
+/**
+ * Process execution result
+ *
+ * @category   Humbug
+ * @package    Humbug
+ * @copyright  Copyright (c) 2015 Pádraic Brady (http://blog.astrumfutura.com)
+ * @license    https://github.com/padraic/humbug/blob/master/LICENSE New BSD License
+ * @author     Thibaud Fabre
+ */
+
+class TestSuiteResult
+{
+
+    const SUCCESS = 0;
+
+    const ERROR = 1;
+
+    /**
+     * @var int Exit code of the test suite.
+     */
+    private $exitCode = 0;
+
+    /**
+     * @var string Capture of the test suite's error output.
+     */
+    private $stdErr = '';
+
+    /**
+     * @var string Capture of the test suite's output.
+     */
+    private $stdOut = '';
+
+    /**
+     * @var int Status code of the test suite execution.
+     */
+    private $status = 0;
+
+    /**
+     * @var CoverageData
+     */
+    private $coverage = null;
+
+    /**
+     * @var float
+     */
+    private $lineCoverage = 0;
+
+    /**
+     * @param PhpProcess $process
+     * @param Container $container
+     * @param string $coverageFileName
+     */
+    public function __construct(PhpProcess $process, Container $container, $coverageFileName)
+    {
```

Comment:
This algorithm should be delegated to some other class. Running logic in constructor is not a good practice.